### PR TITLE
[WIP] Catalyst support for `qml.CosineWindow`

### PR DESCRIPTION
**Pull request by willjmax**
Compiling a circuit with QJIT that begins with a CosineWindow state preparation results in a compiler error. This is because Catalyst has optimizations for when the first operation is a state preparation, but these optimizations are not supported for CosineWindow. A workaround is to only perform the state preparation optimizations on the StatePrep and BasisState classes, rather than the more general StatePrepBase class.

This workaround requires additional changes to Catalyst [(PR #1166)](https://github.com/PennyLaneAI/catalyst/pull/1166) and PennyLane [(PR #6318)](https://github.com/PennyLaneAI/pennylane/pull/6318).

### CODE REVIEW

**Comment by maliasadi** [comment]
@willjmax Will you rebase against the latest changes in the master branch and add an entry to [changelog](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/.github/CHANGELOG.md)?

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml [hunk 1]
```diff
@@ -48,6 +48,7 @@ T                      = { properties = [ "invertible", "controllable", "differe
 # strategy in PennyLane.
 MultiControlledX       = {}
 QFT                    = {}
+CosineWindow           = {}
```
Comment (maliasadi) [inline pennylane_lightning/lightning_qubit/lightning_qubit.toml:51]: :+1:

**Review by maliasadi — COMMENTED** [review]
LGTM!

**Comment by willjmax** [comment]
Closing as this PR is not needed since https://github.com/PennyLaneAI/catalyst/pull/1166 has been merged

**Pull request closed without merging**